### PR TITLE
Add mattermost alert group integration flow

### PR DESCRIPTION
**Pull request by ravishankar15**
Add comment

# What this PR does

## Which issue(s) this PR closes

Related to [issue link here]

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: engine/apps/mattermost/models/channel.py [hunk 1]
```diff
@@ -49,11 +49,33 @@ class Meta:
...
+            alert_group.channel_filter is None
+            or not alert_group.channel_filter.notification_backends
+            or not alert_group.channel_filter.notification_backends.get(MattermostBackend.backend_id)
+        ):
+            return default_channel
+
+        channel_id = alert_group.channel_filter.notification_backends[MattermostBackend.backend_id].get("channel")
+        enabled = alert_group.channel_filter.notification_backends[MattermostBackend.backend_id].get("enabled")
+
+        if not enabled or not channel_id:
```
Comment (matiasb) [inline engine/apps/mattermost/models/channel.py:68]: Shouldn't we avoid sending a Mattermost notification if the setting is not enabled? So you need to explicitly enable and choose a channel (or leave the channel empty to use default).

File: engine/apps/mattermost/tests/test_backend.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "input_data,expected_data",
+    [
+        ({"enabled": True}, {"enabled": True}),
+        ({"enabled": False}, {"enabled": False}),
+        ({"channel": None, "enabled": True}, {"channel": None, "enabled": True}),
+        ({"channel": None}, {"channel": None}),
```
Comment (matiasb) [inline engine/apps/mattermost/tests/test_backend.py:71]: `input_data` and `expected_data` are always the same? I think you are missing some case to cover when you make changes to input during validation? (e.g. passing `"enabled": 1`? some of the other exceptional cases?)

**Review by matiasb — COMMENTED** [review]
LGTM, minor comments added.

File: engine/apps/mattermost/tests/test_backend.py (hunk 2 above)
Comment (ravishankar15) [reply engine/apps/mattermost/tests/test_backend.py:74]: Yeah the idea of this test case is to cover the toggle option flow and the option to clear the channel once it is selected using the `X` on the dropdown. For other exception cases I have added separate tests [ref](https://github.com/grafana/oncall/pull/5347/files#diff-b9c01c665a1c5d4873783e707a541cab535839629a294638c1cde5d894ab8ca3R88). Do you see anyother case that needs to be covered in the tests ?

**Comment by matiasb** [comment]
Thanks!

**Pull request merged**